recipes/magnum/all/test_package/conanfile.py

```diff
@@ -4,18 +4,22 @@
 
 
 class TestPackageConan(ConanFile):
-    settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake", "cmake_find_package_multi"
+
+    def build_requirements(self):
+        if hasattr(self, "settings_build"):
+            self.build_requires(str(self.requires["magnum"]))
```

Comment:
Two comments:
 + Why is this needed?
 + Although this strategy works in CCI, it is essentially wrong. It will require `magnum` for the _build_ context, which is a different configuration from the one that has been built. In a cross-building scenario, the command `conan create` will fail because `magnum` is not built for that configuration.
 
 ```suggestion
```

Reply:
Indeed it's not required since CMakeLists of test package don't test executable imported targets when cross-building, which is probably not very neat: https://github.com/conan-io/conan-center-index/blob/master/recipes/magnum/all/test_package/CMakeLists.txt#L30-L83

Reply:
Yes, it is the same strategy and it is wrong. 

When you execute `conan create --profile:host=windows --profile:build=linux` you create packages for `windows`, and the `test_package/conanfile.py` will receive the same profile inputs. If the `test_package/conanfile.py` uses the package itself as build-requires, then it will be requesting a package for `linux` which hasn't been built. In `test_package/conanfile.py` we can only use/test the packages that have been built (`windows` ones).

In CCI this happens to work because we are building Intel-macos before building M1-macos. The cross-building scenario is `conan create --profile:host=m1 --profile:build=intel`. It works because `intel` has already been built, so the package for `intel` is available and it is found when used as build-requires.

If, for some reason, we decide to trigger `intel` and `m1` in parallel, or we do it the other way around (in the future we will be using M1 machines to build all MacOS packages) these pull-requests will fail. 

Locally, users running intel-macos would need to build first Intel-macos binaries (no cross-building) and then cross-build binaries for M1.... which can be unexpected if you are not aware of these issues.